site/app/controllers/forum/ForumController.php

```diff
@@ -527,6 +531,43 @@ public function publishPost() {
                 $result['next_page'] = $this->core->buildCourseUrl(['forum', 'threads', $thread_id]) . '?' . http_build_query(['option' => $display_option]);
                 $result['post_id'] = $post_id;
                 $result['thread_id'] = $thread_id;
+                $posts = $this->core->getQueries()->getPostsInThreads([$result['thread_id']]);
+                $first = true;
+                $first_post_id = 1;
+                $order_array = [];
+                $reply_level_array = [];
+                if ($display_option != 'tree') {
+                    $reply_level = 1;
+                }
+                else {
+                    foreach ($posts as $post) {
+                        if ($thread_id == -1) {
+                            $thread_id = $post["thread_id"];
+                        }
+                        if ($first) {
+                            $first = false;
+                            $first_post_id = $post["id"];
+                        }
+                        if ($post["parent_id"] > $first_post_id) {
+                            $place = array_search($post["parent_id"], $order_array);
+                            $tmp_array = [$post["id"]];
+                            $parent_reply_level = $reply_level_array[$place];
+                            while ($place !== false && $place + 1 < count($reply_level_array) && $reply_level_array[$place + 1] > $parent_reply_level) {
+                                $place++;
+                            }
+                            array_splice($order_array, $place + 1, 0, $tmp_array);
+                            array_splice($reply_level_array, $place + 1, 0, $parent_reply_level + 1);
+                        }
+                        else {
+                            array_push($order_array, $post["id"]);
+                            array_push($reply_level_array, 1);
+                        }
+                    }
+                }
+                $place = array_search($post_id, $order_array);
+                $reply_level = $reply_level_array[$place];
+                $max_post_box_id = count($posts);
+                $this->sendSocketMessage(['type' => 'new_post', 'thread_id' => $thread_id, 'post_id' => $post_id, 'reply_level' => $reply_level, 'post_box_id' => $max_post_box_id]);
```

Comment:
There are a few lines like this which are very long.  It would be more readable to put each key+value pair on a separate line like this:

```suggestion
                $this->sendSocketMessage([
                    'type' => 'new_post',
                    'thread_id' => $thread_id,
                    'post_id' => $post_id,
                    'reply_level' => $reply_level,
                    'post_box_id' => $max_post_box_id,
                ]);
```